vms/platformvm/state/state.go

```diff
@@ -819,8 +849,37 @@ func (s *state) AddSubnet(createSubnetTx *txs.Tx) {
 }
 
 func (s *state) GetSubnetOwner(subnetID ids.ID) (fx.Owner, error) {
+	if owner, exists := s.subnetOwners[subnetID]; exists {
+		return owner, nil
+	}
+
+	if ownerAndSize, cached := s.subnetOwnerCache.Get(subnetID); cached {
+		if ownerAndSize.size == 0 {
```

Comment:
I feel like this would be a tad more clear
```suggestion
		if ownerAndSize.owner == nil {
```